Autopilot/AttitudeManager/OrientationControl.c

```diff
@@ -1,146 +1,140 @@
-/*
- * File:   OrientationControl.c
- * Author: Chris Hajduk
- *
- * Created on October 29, 2013, 9:41 PM
+/**
+ * @file OrientationControl.c
+ * @author Ian Frosst
+ * @date March 2, 2017
+ * @copyright Waterloo Aerial Robotics Group 2017 \n
+ *   https://raw.githubusercontent.com/UWARG/PICpilot/master/LICENCE 
  */
-#include "OutputCompare.h"
+
 #include "OrientationControl.h"
 #include "main.h"
-#include "AttitudeManager.h"
-#include "VN100.h"
-
-//float constmax = 2.7188;
-
-
-//TODO: Change these variable names to more generic names for inclusion of heading
-//25.2125988006591
-float kd_gain[7] = {3.5, 3.5, 3, 5, 0, 0, 0};
-float kp_gain[7] = {1, 1.6, 4.8, 2, 1.5, 1, 1};//{1, 0.5, 2.5, 1.5, 1.25, 0.05};
-float ki_gain[7]= {0, 0, 0, 0, 0, 0,0};
-//Interal Values
-float sum_gain[6] = {0, 0, 0, 0, 0, 0};
-long int lastControlTime[6] = {0, 0, 0, 0, 0, 0};
-//Derivative Values
-int lastError[6] = {0, 0, 0, 0, 0, 0}; //[0],[1],[2] are currently unused
-
-char integralFreeze = 0;
-char gainsUpdated = 0;
-
-int controlSignalThrottle(int setpoint, int output){
-    int error = setpoint - output;
-    if (integralFreeze == 0){
-        sum_gain[THROTTLE] += (float)error;
-    }
-    int controlSignal = (int)(THROTTLE_SCALE_FACTOR * (error * kp_gain[THROTTLE] + sum_gain[THROTTLE] * ki_gain[THROTTLE]));
-    return controlSignal;
-}
 
-int controlSignalFlap(int setpoint, int output){
-    int error = setpoint - output;
-    if (integralFreeze == 0){
-        sum_gain[FLAP] += (float)error;
-    }
-    int controlSignal = (int)(FLAP_SCALE_FACTOR * (error * kp_gain[FLAP] + sum_gain[FLAP] * ki_gain[FLAP]));
-    return controlSignal;
-}
+/**
+ * Filtering constant for derivative. Between 0 and 1.
+ * The larger this is, the more twitchy D control is. 
+ */
+#define FILTER 0.4f
+
+/* 
+ * Floating-point PID loops for attitude control. 
+ * Inspired by ArduPilot control code.
+ * TODO: convert to fixed-point (integers) for increased speed, decreased size.
+ * (could probably do with tenth- or hundredth-degree integers)
+ * Floating-point math on the dsPIC33 is done via emulation, and is very slow.
```

Comment:
Was looking at options. There's -relaxed-math, with does exactly that, and there's -fast-math, that does even faster floating point math while maintaining precision, but at the cost of size.